utilities/backupable/backupable_db.cc

```diff
@@ -1381,6 +1423,21 @@ Status BackupEngineImpl::CopyOrCreateFile(
   if (s.ok()) {
     s = dest_writer->Close();
   }
+  if (s.ok() && UseNewNaming()) {
+    // When copying SST files and using db_session_id in the name,
+    // try to get DB identities
+    // Note that when CopyOrCreateFile() is called while restoring, this block
+    // of code is not executed as for now we do not use session_id to verify
+    // the restored file
+    if (!src.empty()) {
+      // copying
+      bool isSst = (src.length() > 4 && src.rfind(".sst") == src.length() - 4);
```

Comment:
Could we instead pass in a FileType and check for kTableFile?

Reply:
We could do that. Some preparation work needs to be done in `BackupEngineImpl::Initialize` if passing in a FileType or do the check directly in `BackupEngineImpl::CopyOrCreateFile` without passing in FileType. 

In either case, we will use `ParseFileName` to determine the FileType but I think it just does a bit more than the check here, which also obtains the file number (not necessary here). So l would prefer leaving it as it is.

Reply:
Another option then is to have a shared function for determining the file type and calling that both places. Then it's clear that if checking the file extension is a hack, then it's at least a canonical hack. :)

Reply:
and a consistent hack. :)